cpu/stm32f1/periph/gpio.c

```diff
@@ -30,1015 +30,190 @@
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+/**
+ * @brief   Number of available external interrupt lines
+ */
+#define GPIO_ISR_CHAN_NUMOF             (16U)
+
 typedef struct {
     gpio_cb_t cb;       /**< callback called from GPIO interrupt */
     void *arg;          /**< argument passed to the callback */
-} gpio_state_t;
+} exti_ctx_t;
 
-static gpio_state_t config[GPIO_NUMOF];
+/**
+ * @brief   Allocate memory for on callback and argument per EXTI channel
```

Comment:
This reads strange somehow. Maybe remove "on" or add a minus, e.g.: "on-callback"